devito/finite_differences/finite_difference.py

```diff
@@ -392,3 +447,16 @@ def generate_fd_shortcuts(function):
             derivatives[name_fd] = (deriv, desciption)
 
     return derivatives
+
+def symbolic_weights(function, deriv_order, indices, dim):
+
+    n_weights = len(indices)
+
+    # This 'function' should be centralised
```

Comment:
?

Reply:
The weight function needs to be defined somewhere more appropriate and so that all routines that need it have access to that particular function and don't need to 're-create' it. But since I'm not sure where it's best to put it yet I've left it there (with a reminder for me to change) it for now.

Reply:
oh, I think I see.

Shouldn't it be ``Coefficient('W')`` anyway?

Random thought, maybe BS: perhaps a Function should create a "stamp" Coefficient so that when u do ``function._coefficients`` you get it (instead of a string) ?

Also: should we use ``'W_%s % function.name`` as coefficient name, for uniqueness ?